### PR TITLE
Adds configuration option to adjust the chat fade delay

### PR DESCRIPTION
**Pull request by WesleySykes**
More options can be added
Forever option is not technically forever but lasts longer than any
user should reasonable have the application open

This adds a feature requested in #34

### CODE REVIEW

File: app/renderer/components/settings/sections/Fade.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            this.props.dispatch!(setSetting('fade', parseInt(value)))
+          }}
+        >
+        
+          <option
+            key='10s'
+            value={10000}
+            selected={10000 === this.props.settings.fade}
+          >
+            10 Seconds
```
Comment (AustinCarr) [inline app/renderer/components/settings/sections/Fade.tsx]: "Seconds" should probably come from the localization text.

File: app/renderer/components/settings/sections/Fade.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          >
+            30 Seconds
+          </option>
+
+          <option
+            key='forever'
+            value={9999999999999999}
+            selected={9999999999999999 === this.props.settings.fade}
+          >
+            Forever
```
Comment (AustinCarr) [inline app/renderer/components/settings/sections/Fade.tsx]: "Forever" should also probably come from the localization text.

**Comment by WesleySykes** [comment]
Oops, I didn't think about the localization there.

**Comment by WesleySykes** [comment]
I added localization for 'Forever' and 'Seconds'. I wasn't sure whether it would be better to have localization for just the word 'Seconds' or to make separate localization for each option. I opted for adding one for each option because potentially other languages could work differently (i.e. add the word for seconds before the actual number) and because it removes any hard coded text from the Fade.tsx file but it can be changed if another way is better.

**Comment by AustinCarr** [comment]
I can't make any changes to the code at the moment, so I'll just type out some thoughts on the direction I would go in:

You've removed the hardcoded text from the Fade.tsx file, which is good! But it's still hardcoded in the localization file.

What if people translate the text to a bunch of different languages and then we want to add a "5 Seconds" option? We'd have to go back and change all the localization files to account for that. That's kind of annoying, but not that big of a deal.

But, what if somebody wants the chat to fade after 7 seconds? Do we add a "7 seconds" dropdown option? Nah, that's too many options. It'd be best at that point to make the fade time selection an input box instead of a dropdown. So, how do we write translations for _that_ scenario?

We need to make the translation dynamic! We need to use [interpolation](https://www.i18next.com/translation-function/interpolation).

Take a look at the `shareFriendCodeOther` key in the `en-US.ts` file. It has a value of ` Share {{name}}’s friend code below to invite friends.`, where `{{name}}` is dynamic.

You can see this used in the file `Invite.tsx`. The code looks like this: `t('shareFriendCodeOther', { name: this.props.hostName })`.

**Comment by WesleySykes** [comment]
I see. Thank you for the insight. I will look into this soon and see if I can improve it.

**Comment by WesleySykes** [comment]
I was able to convert it to a number box instead of a drop down.

Some parts were a little tricky but I hope this is sufficient.

**Comment by samuelmaddock** [comment]
Hey @WesleySykes, I'm making some [major changes to Metastream](https://www.patreon.com/posts/update-on-25999308) and might need to hold off on merging these changes for awhile. Hope that's alright with you.

**Comment by WesleySykes** [comment]
Okay that's fine. :)

**Comment by samuelmaddock** [comment]
@WesleySykes You're welcome to reapply these changes to the new project on the `master` branch now.

**Comment by samuelmaddock** [comment]
I'm going to close this PR for now, but feel free to open it again when changes are made.

**Comment by WesleySykes** [comment]
I apologize for the late reply. I have been a little busy and haven't been able to keep up with the recent changes. What should I do to merge my changes to the most recent version?

**Comment by samuelmaddock** [comment]
@WesleySykes You might be able to rebase your changes and have them apply to the new files. There will likely be conflicts that you'll need to resolve. Otherwise you could manually re-add changes to the new code.

**Pull request closed without merging**